src/main/scala/uk/gov/nationalarchives/StatusProcessor.scala

```diff
@@ -0,0 +1,101 @@
+package uk.gov.nationalarchives
+
+import cats.Monad
+import cats.implicits._
+import uk.gov.nationalarchives.Lambda.{File, Input, Status}
+import uk.gov.nationalarchives.PuidRepository.AllPuidInformation
+
+import scala.util.Try
+
+class StatusProcessor[F[_] : Monad](input: Input, allPuidInformation: AllPuidInformation) {
+  private val Success = "Success"
+  private val VirusDetected = "VirusDetected"
+  private val Antivirus = "Antivirus"
+  private val NonJudgmentFormat = "NonJudgmentFormat"
+  private val Mismatch = "Mismatch"
+  private val ChecksumMatch = "ChecksumMatch"
+  private val FileType = "File"
+  private val Failed = "Failed"
+  private val ServerChecksum = "ServerChecksum"
+  private val ZeroByteFile = "ZeroByteFile"
+  private val ClientChecksum = "ClientChecksum"
+  private val ClientFilePath = "ClientFilePath"
+  private val FFIDStatus = "FFID"
+
+  def antivirus(): F[List[Status]] = {
```

Comment:
Is it worth create some individual unit tests for these methods? The internal logic is important, and for the ffid statuses quite complex to warrant some thorough testing?

Reply:
I thought about it. The reason I didn't is because the `LambdaTest` tests cover each of the paths. I've run the code coverage in IntelliJ and the tests are hitting every line in `StatusProcessor` The lambda tests will catch any changes made to the DisallowedPuids SQL as well and I think generally, integration tests on their own are OK for lambdas.

I can put extra test cases into the LambdaTest though if there's anything missing.